frontend/components/suggestion-and-search.js

```diff
@@ -0,0 +1,134 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import debounce from 'lodash.debounce';
+import { Badge } from './badge-examples';
+
+function toXhrSend(data) {
+  let str = '', start = true;
+  let jsondata = '';
+  for (const key in data) {
+    if (typeof (jsondata = JSON.stringify(data[key])) === 'string') {
+      str += start ? '' : '&';
+      if (typeof data[key] === 'string') {
+        str += encodeURIComponent(key) + '=' + encodeURIComponent(data[key]);
+      } else {
+        str += encodeURIComponent(key) + '=' + encodeURIComponent(jsondata);
+      }
+      start = false;
+    }
+  }
+  return str;
+}
+
+function ajax(baseUri, verb, adverbs, cb) {
```

Comment:
I wonder if we should pull this out into a separate `ajax.js` file, as it's useful outside this particular file.